src/components/CellsContainer/CellsContainer.svelte

```diff
@@ -0,0 +1,201 @@
+        .on('mouseover', handleRectHover)
+        .on('mouseout', handleRectMouseout)
+        .merge(rects)
+        .attr('fill', keyGetter('color'))
+        .attr('rx', 3)
+        .attr('ry', 3)
+        .attr('x', keyGetter('x'))
+        .attr('y', keyGetter('y'))
+        .attr('width', keyGetter('size'))
+        .attr('height', keyGetter('size'))
+        .style('cursor', 'pointer');
+    }
+  }
+
+  function handleRectHover(e, rect) {
+    select(this)
+      .transition()
+      .attr('fill', keyGetter('hoverColor'))
+      .duration(150);
+
+    if (rect.tooltipText) {
+      createTooltip(rect);
+    }
+  }
+
+  function createTooltip(rect) {
+    const xPadding = 5;
+    const yPadding = 2;
+    const textY = rect.y - 8;
+    const textX = rect.x + xPadding;
+
+    const [bgId, textId, triangleId] = [
+      getId('bg', rect.index),
+      getId('text', rect.index),
+      getId('triangle', rect.index)
+    ];
+
+    const bg = g.append('rect');
+
+    const text = g.append('text').text(rect.tooltipText);
+
+    const triangle = g.append('path');
+
+    setAttributes(text, {
+      id: textId,
+      fill: tooltipColor,
+      x: textX,
+      y: textY,
+      'font-size': '12px'
+    });
+
+    setAttributes(bg, { id: bgId, fill: tooltipBackground, rx: 3, ry: 3 });
+
+    setAttributes(triangle, {
+      fill: tooltipBackground,
+      id: triangleId,
+      d: `M ${rect.x + rect.size / 2} ${rect.y} l 3 -3 l -6 0 z`
+    });
+
+    const { width, height } = text.node().getBoundingClientRect();
+    const bgWidth = width + xPadding * 2;
+
+    setAttributes(bg, {
+      x: rect.x,
+      y: textY - height + yPadding,
+      width: bgWidth,
+      height: height + yPadding * 2
+    });
+
+    if (Number(bg.attr('y')) + Number(margin.top) < 0) {
+      //outside top
+      const margin = rect.y + rect.size + 2;
+      bg.attr('y', margin);
+      text.attr('y', margin + height - yPadding);
+      triangle.attr(
+        'd',
+        `M ${rect.x + rect.size / 2} ${rect.y + rect.size} l -3 3 l 6 0 z`
+      );
+    }
+
+    // centrating
+    if (bgWidth > rect.size) {
+      const offset = (bgWidth - rect.size) / 2;
+      bg.attr('x', rect.x - offset);
+      text.attr('x', textX - offset);
+    } else if (bgWidth < rect.size) {
+      const offset = (rect.size - bgWidth) / 2;
+      bg.attr('x', rect.x + offset);
+      text.attr('x', textX + offset);
+    }
+
+    const bgOuterLeft = Number(bg.attr('x')) + Number(margin.left);
+    const bgOuterRight = bgOuterLeft + bgWidth - margin.right;
+
+    if (bgOuterRight > innerWidth + margin.right) {
+      //outside right
+      const offset = bgOuterRight - innerWidth + xPadding * 2 - margin.right;
+      bg.attr('x', bg.attr('x') - offset);
+      text.attr('x', text.attr('x') - offset);
+    } else if (bgOuterLeft < 0) {
+      //outside left
+      bg.attr('x', -margin.left);
+      text.attr('x', xPadding - margin.left);
+    }
+  }
+
+  function removeTooltip(index) {
```

Comment:
and functions again )
If you are using such structure, would be nice to have some comments in the code about what a particular section is doing